app/Domain/Contacts/Models/ContactModel.php

```diff
@@ -32,7 +42,10 @@ final class ContactModel extends UuidModel
     protected static function booted(): void
     {
         self::created(
-            static fn ($contactModel) => ContactCreated::dispatch(Uuid::fromString($contactModel->id))
+            static fn (ContactModel $contactModel) => ContactCreated::dispatch($contactModel->getId())
+        );
+        self::updated(
+            static fn (ContactModel $contactModel) => ContactUpdated::dispatch($contactModel->getId())
```

Comment:
Nit: You could also do `$contactModel->toDomain()->id` and then you don't need the extra methods

Reply:
Good point, they shouldn't be used anyway, only from the domain model so it makes sense.